### ISSUE TITLE
no-throw-literal has false positive at await expression

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#7686]
**Tell us about your environment**

* **ESLint Version:** 3.11.1
* **Node Version:** 6.9.1
* **npm Version:** 3.8.9

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

```
$ eslint test.js --parser-options "ecmaVersion:2017" --rule "no-throw-literal:error" --no-eslintrc
```

```js
async function foo() {
    throw await bar()
}
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
test.js
  3:5  error  Expected an object to be thrown  no-throw-literal

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
I will work on this after #7689 to avoid confliction.
Or I'm OK if #7689 fixes this issue :smile:

**Comment by not-an-aardvark** [comment]
I pushed a commit to https://github.com/eslint/eslint/pull/7689 that fixes this issue.

**Comment by mysticatea** [comment]
Thank you!

**Comment by mysticatea** [comment]
Closing as this was fixed by #7689.
